### ISSUE TITLE
r0.11 iOS build and examples fail with duplicate symbols in libtensorflow-core.a

### ISSUE THREAD

**Issue by avalys** [tensorflow/tensorflow#5159]
This is from a fresh checkout of commit-id 40d28e24df77130792dc20408e45dda3154b9bc2, the current r0.11 branch.

On Mac OS 10.11.6 and Xcode 8.0, building the iOS library with build_ios_all.sh completes without errors, but attempting to actually build the Xcode projects in contrib/ios_examples or use the resulting libtensorflow-core.a in my own project fails with a linker error.

```
duplicate symbol __Z14tf_git_versionv in:
    /Users/avalys/Desktop/tensorflow.build_ios/tensorflow/contrib/ios_examples/simple/../../makefile/gen/lib/libtensorflow-core.a(version_info.o)
duplicate symbol __Z19tf_compiler_versionv in:
    /Users/avalys/Desktop/tensorflow.build_ios/tensorflow/contrib/ios_examples/simple/../../makefile/gen/lib/libtensorflow-core.a(version_info.o)
ld: 2 duplicate symbols for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

And indeed, nm -a libtensorflow-core.a | grep git_version gives:

```
0000000000000000 T __Z14tf_git_versionv
0000000000000000 T __Z14tf_git_versionv
```

**Comment by avalys** [comment]
This was apparently fixed in `master` by a change to `tensorflow/contrib/makefile/Makefile`:  https://github.com/tensorflow/tensorflow/commit/1283b84a49a9f5e14aca833cf981b61848aaf916#diff-acde23d6b74f5e8c54a951f435364be7R428

version_info.cc would seem to be included twice.

**Comment by asimshankar** [comment]
@avalys : Sorry for the trouble and thanks very much for tracking down the fix yourself.

I'm going to close this out since it seems you've found the right resolution. Feel free to reopen if I have misunderstood or if your problem hasn't been addressed.

We'll look into possibly cherry picking this into the next release candidate for 0.11, but no promises yet. We tend to avoid delayed releases to fix issues in the contrib/ directory.

**Comment by avalys** [comment]
This is such a simple fix, I'd hate to see 0.11 get released without a working iOS build.

There is a bunch of other stuff in that commit - I've submitted a pull request with just this fix if it makes it any easier.

https://github.com/tensorflow/tensorflow/pull/5173

**Comment by asimshankar** [comment]
@avalys : This fix will be part of 0.11.0rc2 (and thus 0.11 final)
